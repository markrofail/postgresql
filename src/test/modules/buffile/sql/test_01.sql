-- This is the first test, so make sure the test extension is there.
CREATE EXTENSION IF NOT EXISTS buffile;

BEGIN;
SELECT buffile_create();
SELECT buffile_seek(0, 1);
SELECT buffile_write('abc');
SELECT buffile_seek(0, 0);
-- Check that the trailing zeroes are not fetched.
SELECT buffile_read(16);
-- Adjust the number of useful bytes.
SELECT buffile_write('abc');
-- ... and check again.
SELECT buffile_seek(0, 0);
SELECT buffile_read(16);
SELECT buffile_close();
COMMIT;
