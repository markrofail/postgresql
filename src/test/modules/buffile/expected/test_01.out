-- This is the first test, so make sure the test extension is there.
CREATE EXTENSION IF NOT EXISTS buffile;
BEGIN;
SELECT buffile_create();
 buffile_create 
----------------
 
(1 row)

SELECT buffile_seek(0, 1);
 buffile_seek 
--------------
            0
(1 row)

SELECT buffile_write('abc');
 buffile_write 
---------------
             3
(1 row)

SELECT buffile_seek(0, 0);
 buffile_seek 
--------------
            0
(1 row)

-- Check that the trailing zeroes are not fetched.
SELECT buffile_read(16);
 buffile_read 
--------------
 \x00616263
(1 row)

-- Adjust the number of useful bytes.
SELECT buffile_write('abc');
 buffile_write 
---------------
             3
(1 row)

-- ... and check again.
SELECT buffile_seek(0, 0);
 buffile_seek 
--------------
            0
(1 row)

SELECT buffile_read(16);
   buffile_read   
------------------
 \x00616263616263
(1 row)

SELECT buffile_close();
 buffile_close 
---------------
 
(1 row)

COMMIT;
